src/DataLayer/dataSources/tests/DataSource.spec.ts

```diff
@@ -32,8 +36,140 @@ describe('DataSource', () => {
+            return { responseFactory, resultFactory, dataSource };
+        };
+
+        it('should throw when missing responseFactory', () => {
+            const execFactory = jest.fn().mockReturnValue(Promise.resolve());
+            const responseFactory = undefined;
+            const resultFactory = jest.fn();
+
+            const dataSource = new DataSource(execFactory, afm, 'finger', responseFactory, resultFactory);
+            expect(() => {
+                dataSource.getPage(resultSpec);
+            }).toThrow();
+        });
+
+        it('should throw when missing resultFactory', async () => {
+            const execFactory = jest.fn().mockReturnValue(Promise.resolve());
+            const responseFactory = jest.fn().mockReturnValue(Promise.resolve({ links: { executionResult: 'url1' } }));
+            const resultFactory = undefined;
+
+            const dataSource = new DataSource(execFactory, afm, 'finger', responseFactory, resultFactory);
+            try {
+                await dataSource.getPage(resultSpec);
+            } catch (e) {
+                expect(e).toBeInstanceOf(Error);
+            }
+        });
+
+        it('should use responseFactory+resultFactory', async () => {
+            const { responseFactory, resultFactory, dataSource } = createDataSource();
+            const responses = await dataSource.getPage(resultSpec);
+
+            expect(responseFactory).toBeCalledWith(resultSpec);
+            expect(resultFactory).toBeCalledWith('url1', [0, 0], [SERVER_SIDE_ROW_LIMIT, SERVER_SIDE_COLUMN_LIMIT]);
+            expect(responses).toEqual({
+                executionResponse: { links: { executionResult: 'url1' } },
+                executionResult: { data: [] }
+            });
+        });
+
+        it('should cache first response', async () => {
+            const { responseFactory, resultFactory, dataSource } = createDataSource();
+            await dataSource.getPage(resultSpec);
+            const responses2 = await dataSource.getPage(
+                resultSpec,
+                [SERVER_SIDE_ROW_LIMIT, SERVER_SIDE_COLUMN_LIMIT],
+                [10, 10]
+            );
+
+            expect(responseFactory).toHaveBeenCalledTimes(1);
+            expect(resultFactory).toHaveBeenCalledTimes(2);
+            expect(resultFactory.mock.calls[0]).toEqual([
+                'url1',
+                [0, 0],
+                [SERVER_SIDE_ROW_LIMIT, SERVER_SIDE_COLUMN_LIMIT]
+            ]);
+            expect(resultFactory.mock.calls[1]).toEqual([
+                'url1',
+                [SERVER_SIDE_ROW_LIMIT, SERVER_SIDE_COLUMN_LIMIT],
+                [10, 10]
+            ]);
+            expect(responses2).toEqual({
+                executionResponse: { links: { executionResult: 'url1' } },
+                executionResult: { data: [[]] }
+            });
+        });
+
+        it('should work with different resultSpec', async () => {
```

Comment:
what does it mean `work with different resultSpec` ?

Reply:
That it should be able to handle two different result spec requests

Reply:
`should be able to handle two different result spec requests` this is better message than `should work with different resultSpec` :) could you please update it ? :)